### PR TITLE
web: flag explanation: make more readable

### PR DESCRIPTION
**Pull request by lainisourgod**
* add spacing between separate flag explanations
* make flag name bold
* make explanation property array of Text-s so that flag name and explanation may be styled differently

I'm little new to JSX so I got multiple type errors that I don't know how to resolve, but everything works just fine. 

CleanShot 2023-02-04 at 20 53 29@2x

I want in ExplainProps receive not a `string` array, but array of `Text`s that are styled differently.

### CODE REVIEW

File: website/src/components/Explain.tsx
Comment (notmd) [inline website/src/components/Explain.tsx:15]: Maybe replace this with ReactElement[] would work

**Review by notmd — COMMENTED** [review]
I haven't look the visual yet

File: website/src/components/Explain.tsx
Comment (notmd) [inline website/src/components/Explain.tsx:29]:
```suggestion
            <Fragment key={idx}>
```

**Review by notmd — CHANGES_REQUESTED** [review]

File: website/src/components/Explain.tsx
Comment (fozziethebeat) [reply website/src/components/Explain.tsx:15]: Yeah, i think this will fix the build issue.

**Review by fozziethebeat — CHANGES_REQUESTED** [review]
Looks good once the build works

File: website/src/components/Explain.tsx
Comment (lainisourgod) [reply website/src/components/Explain.tsx:29]: I tried changing `<></>` to `Fragment`, linter doesn't like it (1).
```tsx
          {explanation.map((paragraph, idx) =>
            <Fragment key={idx}>
              {paragraph}
              <br />
            </Fragment>
          )}
```

CleanShot 2023-02-05 at 17 51 14@2x

If I remove `<br />`, it now doesn't like the `key` property (2)
CleanShot 2023-02-05 at 17 53 07@2x

If I remove `key` property, everything is fine again.

CleanShot 2023-02-05 at 17 54 09@2x

I'm not really sure what is going on here.

**Comment by lainisourgod** [comment]
Also @notmd I rebase my code on top of yesterday's commits and now another type issue appeared out of nowhere.
CleanShot 2023-02-05 at 17 54 47@2x

VSCode suggests to put fake children property, I'm not sure it's a good fix.
CleanShot 2023-02-05 at 17 55 58@2x

**Comment by lainisourgod** [comment]
Also I forgot to pin the screenshot of how it looks now. 

image

File: website/src/components/Explain.tsx
Comment (notmd) [reply website/src/components/Explain.tsx:29]: hey @lainisourgod, sorry for the bad code suggestion, please try this:
```jsx
<Fragment key={idx}>
  <>
    {paragraph}
    <br />
  </>
</Fragment>
```

Comment (notmd) [reply website/src/components/Explain.tsx:29]: Hmm, look like the TS doesn't complain when use change the type to ReactElement[]

**Comment by notmd** [comment]
> Also @notmd I rebase my code on top of yesterday's commits and now another type issue appeared out of nowhere. CleanShot 2023-02-05 at 17 54 47@2x
> 
> VSCode suggests to put fake children property, I'm not sure it's a good fix. CleanShot 2023-02-05 at 17 55 58@2x

Have you tried restarting VSCode? On my end, it doesn't complain anything about it.

File: website/src/components/Explain.tsx
Comment (lainisourgod) [reply website/src/components/Explain.tsx]: I'm not really proficient in TS type system, but using ReactElement here feels a hack, but if it's ok for your code base, then fine!

**Comment by lainisourgod** [comment]
So now another changed happend during the day and now Explanation API is used in other code, so my changes are conflicting with this usage. I wonder if should change the Explanation API back how it was and just render `<br />` between lines without **bold**, or should I update API usage in `LabelLikertGroup.tsx`.

I should check if rendering label name and description seperately is beneficial in `Likert` case, and if it so, change the usage in `LabelLikertGroup.tsx`

**Comment by lainisourgod** [comment]
Hi @notmd! Sorry for not being active on issue last week.

I removed `bold` part of changes "**Wrong language:** Not written ..." so not to make API unnecessary complicated and leaved only line breaks.

Also I noticed templating issues likely introduced by #1349 which I fixed by copying templating realization from this PR: https://github.com/LAION-AI/Open-Assistant/blob/main/website/src/components/Messages/LabelFlagGroup.tsx#L44
CleanShot 2023-02-16 at 01 59 52@2x

**Comment by olliestanley** [comment]
@notmd @fozziethebeat will both have to review this and confirm that changes requested have been addressed before it can be merged

**Review by notmd — APPROVED** [review]
Thanks

**Comment by notmd** [comment]
@olliestanley I just update the PR, can you please force merge if the CI passes?

**Comment by olliestanley** [comment]
> @olliestanley I just update the PR, can you please force merge if the CI passes?

I don't have force merge perms, I think will need Fozzie to approve unless @andreaskoepf has time to force merge

**Pull request merged**
